ecosystem/typescript/sdk_v2/src/core/hex.ts

```diff
@@ -0,0 +1,192 @@
+   *
+   * @returns Hex data as Uint8Array
+   */
+  toUint8Array(): Uint8Array {
+    return this.data;
+  }
+
+  /**
+   * Get the hex data as a string without the 0x prefix.
+   *
+   * @returns Hex string without 0x prefix
+   */
+  toStringWithoutPrefix(): string {
+    return bytesToHex(this.data);
+  }
+
+  /**
+   * Get the hex data as a string with the 0x prefix.
+   *
+   * @returns Hex string with 0x prefix
+   */
+  toString(): string {
+    return `0x${this.toStringWithoutPrefix()}`;
+  }
+
+  /*
+   * Methods for creating an instance of Hex from other types.
+   */
+
+  /**
+   * Static method to convert a hex string to Hex
+   *
+   * @param str A hex string, with or without the 0x prefix
+   *
+   * @returns Hex
+   */
+  static fromString(args: { str: string }): Hex {
+    let input = args.str;
+
+    if (input.startsWith("0x")) {
+      input = input.slice(2);
+    }
+
+    if (input.length === 0) {
+      throw new ParsingError(
+        "Hex string is too short, must be at least 1 char long, excluding the optional leading 0x.",
+        HexInvalidReason.TOO_SHORT,
+      );
+    }
+
+    if (input.length % 2 !== 0) {
+      throw new ParsingError("Hex string must be an even number of hex characters.", HexInvalidReason.INVALID_LENGTH);
+    }
+
+    try {
+      return new Hex({ data: hexToBytes(input) });
+    } catch (e) {
+      const error = e as Error;
+      throw new ParsingError(
+        `Hex string contains invalid hex characters: ${error.message}`,
+        HexInvalidReason.INVALID_HEX_CHARS,
+      );
+    }
+  }
+
+  /**
+   * Static method to convert an instance of HexInput to Hex
+   *
+   * @param str A HexInput (string or Uint8Array)
+   *
+   * @returns Hex
+   */
+  static fromHexInput(args: { hexInput: HexInput }): Hex {
+    if (args.hexInput instanceof Uint8Array) return new Hex({ data: args.hexInput });
+    return Hex.fromString({ str: args.hexInput });
+  }
+
+  /*
+   * Static helper methods below this point. These help the dev avoid the need to build
+   * a Hex instance if they want to convert between different representations of hex.
+   */
+
+  /**
+   * Static method to convert HexInput to bytes
+   *
+   * @returns hex as bytes
+   */
+  static hexInputToBytes(args: { hexInput: HexInput }): Uint8Array {
+    const hex = Hex.fromHexInput(args);
+    return hex.toUint8Array();
+  }
+
+  /**
+   * Static method to convert HexInput to a string with the 0x prefix.
+   *
+   * @returns hex as string with 0x prefix
+   */
+  static hexInputToString(args: { hexInput: HexInput }): string {
+    const hex = Hex.fromHexInput(args);
+    return hex.toString();
+  }
+
+  /**
+   * Static method to convert HexInput to a string without the 0x prefix.
+   *
+   * @returns hex as string without 0x prefix
+   */
+  static hexInputToStringWithoutPrefix(args: { hexInput: HexInput }): string {
+    const hex = Hex.fromHexInput(args);
+    return hex.toStringWithoutPrefix();
+  }
+
```

Comment:
Wondering, why have these as static, if they can just be exactly the same as creating, then output?

Reply:
Discussing offline.